spec/controllers/oauth/applications_controller_spec.rb

```diff
@@ -227,6 +232,73 @@ module Oauth
       expect(response).to have_http_status :forbidden
     end
 
+    it "should let an admin create an application with Oauth Admins" do
+      controller.sign_in! admin
+      post(:create,
+        params: {
+            doorkeeper_application: {
+              name: 'Some app',
+              redirect_uri: 'https://www.example.com',
+              can_message_users: true,
+            },
+            member_ids: user2.id
+        }
+      )
+
+      id = assigns(:application).id
+      expect(id).not_to be_nil
+      expect(response).to redirect_to(oauth_application_path(id))
+      expect(assigns(:application).name).to eq('Some app')
+      expect(assigns(:application).redirect_uri).to eq('https://www.example.com')
+      expect(assigns(:application).can_message_users).to eq(true)
+      expect(assigns(:application).owner.member_ids).to include(user2.id)
+    end
+
+    it "should let an oauth admin edit an application" do
+      controller.sign_in! user
+      get(:edit, params: { id: trusted_application_admin.id })
+
+      expect(response).to have_http_status :success
+      expect(assigns(:application).redirect_uri).to eq(trusted_application_admin.redirect_uri)
+    end
+
+    it "should let an oauth admin update an application" do
+      controller.sign_in! user
+      put(:update,
+        params: {
+          id: trusted_application_admin.id,
+          doorkeeper_application: {
+            redirect_uri: 'https://www.example.org'
+          }
+        }
+      )
+      expect(response).to redirect_to(oauth_application_path(trusted_application_admin.id))
+      expect(assigns(:application).redirect_uri).to eq('https://www.example.org')
+    end
```

Comment:
Can you add a spec showing that oauth admins cannot do certain updates? (all the things that weren't permitted in the controller)

Reply:
Test added